packages/dashboard/src/pages/CreateMaskWallet/CreateMnemonic/index.tsx

```diff
@@ -181,15 +181,16 @@ const CreateMnemonic = memo(function CreateMnemonic() {
         })
     }, [location.state?.password, location.state?.isReset])
 
+    const { value: hasPassword, loading: loadingHasPassword } = useAsync(WalletServiceRef.value.hasPassword, [])
+
     const { value: address } = useAsync(async () => {
         if (!words.length) return
 
-        const hasPassword = await WalletServiceRef.value.hasPassword()
         if (!hasPassword) await WalletServiceRef.value.setDefaultPassword()
 
         const address = await WalletServiceRef.value.generateAddressFromMnemonicWords(walletName, words.join(' '))
         return address
-    }, [words.join(' '), walletName])
+    }, [words.join(' '), walletName, hasPassword])
```

Comment:
`words` is already memoized, no need to serialize it.

```suggestion
    }, [words, walletName, hasPassword])
```

Besides, let's make it more stable:

```diff
diff --git a/packages/dashboard/src/hooks/useMnemonicWordsPuzzle.ts b/packages/dashboard/src/hooks/useMnemonicWordsPuzzle.ts
index 43eac0141..9b6436253 100644
--- a/packages/dashboard/src/hooks/useMnemonicWordsPuzzle.ts
+++ b/packages/dashboard/src/hooks/useMnemonicWordsPuzzle.ts
@@ -3,6 +3,7 @@ import { useAsyncRetry } from 'react-use'
 import { produce } from 'immer'
 import { range, shuffle, remove, clone } from 'lodash-es'
 import { WalletServiceRef } from '@masknet/plugin-infra/dom'
+import { EMPTY_LIST } from '@masknet/shared-base'

 const PUZZLE_SIZE = 3

@@ -15,7 +16,7 @@ export interface PuzzleWord {
 }

 export function useMnemonicWordsPuzzle() {
-    const { value: words = [], retry: wordsRetry } = useAsyncRetry(
+    const { value: words = EMPTY_LIST, retry: wordsRetry } = useAsyncRetry(
         () => WalletServiceRef.value.createMnemonicWords(),
         [],
     ) 
```

Reply:
Put `words` directly in dependency will crash the page.

Reply:
What if with above patch that makes it stable?.

By now, it's OK to me if you still want to serialize it.

Reply:
Amazing, it no longer crash.